tests/conftest.py

```diff
@@ -1,16 +1,114 @@
 import pytest
-from functools import wraps
 
-from ethereum.tools import tester
+# from functools import wraps
+import eth_tester
+import web3
+
+from eth_tester import (
+    EthereumTester,
+)
+from eth_tester.exceptions import (
+    TransactionFailed
+)
+from web3.providers.eth_tester import (
+    EthereumTesterProvider,
+)
+
+from web3 import (
+    Web3,
+)
+from web3.contract import (
+    ConciseContract,
+)
+from web3.exceptions import (
+    ValidationError
+)
 from vyper.parser.parser_utils import (
     LLLnode
 )
 from vyper import (
     compile_lll,
-    optimizer,
     compiler,
+    optimizer,
+)
+from vyper.utils import (
+    sha3
 )
-from ethereum import utils as ethereum_utils
+
+############
+# PATCHING #
+############
+
+
+def Filter_remove(self, *values):
+
+    def get_key(v):
+        return v.get('transaction_hash'), v.get('log_index'), v.get('transaction_index')
+
+    values_to_remove = set([
+        get_key(value)
+        for value in values
+    ])
+
+    queued_values = self.get_changes()
+    self.values = [
+        value
+        for value
+        in self.get_all()
+        if get_key(value) not in values_to_remove
+    ]
+    for value in queued_values:
+        if get_key(value) in values_to_remove:
+            continue
+        self.queue.put_nowait(value)
+eth_tester.utils.filters.Filter.remove = Filter_remove
+
+
+original_is_encodable = web3.utils.abi.is_encodable
+def utils_abi_is_encodable(_type, value):
+    from eth_utils import is_integer
+    from eth_abi.abi import process_type
+    try:
+        base, sub, arrlist = _type
+    except ValueError:
+        base, sub, arrlist = process_type(_type)
+
+    if not arrlist:
+        if base == 'fixed' and not arrlist:
+            return True
+        elif base == 'int':
+            if not is_integer(value):
+                return False
+            exp = int(sub)
+            if value < -1 * 2**(exp - 1) or value > 2**(exp - 1) + 1:
+                return False
+            return True
+
+    # default behaviour
+    return original_is_encodable(_type, value)
+web3.utils.abi.is_encodable = utils_abi_is_encodable
+
+
+@pytest.fixture(scope="module")
+def tester():
+    t = EthereumTester()
+    return t
+
+
+def zero_gas_price_strategy(web3, transaction_params=None):
+    return 0  # zero gas price makes testing simpler.
+
+
+@pytest.fixture(scope="module")
+def w3(tester):
+    w3 = Web3(EthereumTesterProvider(tester))
+    w3.eth.setGasPriceStrategy(zero_gas_price_strategy)
+    return w3
+
+
+@pytest.fixture
+def sha3():
```

Comment:
I'd recommend renaming this to `keccak`.  (which makes me realize we should deprecate `Web3.sha3` and rename it to `keccak` as well.